circuitbreaker.py

```diff
@@ -15,28 +15,81 @@
 except ImportError:
     from monotonic import monotonic
 
+# Python2 vs Python3 strings
+try:
+    basestring
+    STRING_TYPES = (basestring, unicode)
+except NameError:
+    STRING_TYPES = (bytes, str)
+
+
 STATE_CLOSED = 'closed'
 STATE_OPEN = 'open'
 STATE_HALF_OPEN = 'half_open'
 
+def in_exception_list(*exc_types):
+    """Build a predicate function that checks if an exception is a subtype from a list"""
+    def matches_types(thrown_type, thrown_value):
+        return issubclass(thrown_type, exc_types)
+    return matches_types
 
 class CircuitBreaker(object):
     FAILURE_THRESHOLD = 5
     RECOVERY_TIMEOUT = 30
-    EXPECTED_EXCEPTION = Exception
     FALLBACK_FUNCTION = None
 
     def __init__(self,
                  failure_threshold=None,
                  recovery_timeout=None,
                  expected_exception=None,
                  name=None,
-                 fallback_function=None):
+                 fallback_function=None
+                 ):
+        """
+        Construct a circuit breaker.
+
+        :param failure_threshold: break open after this many failures
+        :param recovery_timout: close after this many seconds
```

Comment:
little typo "timeout"
```suggestion
        :param recovery_timeout: close after this many seconds
```